ui/components/component-library/button-icon/button-icon.types.ts

```diff
@@ -0,0 +1,49 @@
+import type { BoxProps } from '../../ui/box/box.d';
+import { IconName, IconSize } from '../icon';
+import { Color, IconColor } from '../../../helpers/constants/design-system';
+
+export enum ButtonIconSize {
+  Sm = IconSize.Sm,
+  Lg = IconSize.Lg,
+}
+
+export interface ButtonIconProps extends BoxProps {
+  /**
+   * String that adds an accessible name for ButtonIcon
+   */
+  ariaLabel: string;
+  /**
+   * The polymorphic `as` prop allows you to change the root HTML element of the Button component between `button` and `a` tag
+   */
+  as?: keyof JSX.IntrinsicElements;
```

Comment:
Maybe if we are more strict on what the root html element can be we can have better control over what html attributes can be passed to the `ButtonIcon` component. Let's prevent changing the root element from anything but a `button` | `as`? I can't think of any reason that you would need to change it to anything else?

```suggestion
  as?: 'button' | 'a';
```